dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreRecordCollection.cs

```diff
@@ -164,6 +170,13 @@ public AzureCosmosDBNoSQLVectorStoreRecordCollection(
             .Concat([properties.KeyProperty])
             .Select(x => this._storagePropertyNames[x.DataModelPropertyName])
             .ToList();
+
+        this._vectorProperties = properties.VectorProperties;
+
+        if (this._vectorProperties.Count > 0)
+        {
+            this._firstVectorProperty = this._vectorProperties.First();
```

Comment:
Knowing it's a list, this can be faster.

```suggestion
            this._firstVectorProperty = this._vectorProperties[0];
```